### ISSUE TITLE
shecllcheck doesn't give warning for included file 

### ISSUE THREAD

**Issue by coiby** [koalaman/shellcheck#2617]
#### For bugs

- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): version: 0.8.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit (no applicable since multiple files are involved)

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this  (no applicable since multiple files are involved)
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

Let's say we have two files
 1. test.sh
```sh
#!/bin/bash
source include.sh
```
 2. include.sh
```
#!/bin/bash
aa=NEWS
bb=test2
```

I want shellcheck to be able to find any issue in `include.sh` but also avoiding any false positive. Correct me if I am wrong, I think the right approach is to make shellcheck emit warnings about the included file when checking `test.sh`  but without directly checking the included file.

#### Here's what shellcheck currently says:

```sh
$ shellcheck  -x test.sh 

$ shellcheck  include.sh 

In include.sh line 2:
aa=NEWS
^-- SC2034 (warning): aa appears unused. Verify use (or export if used externally).


In include.sh line 3:
bb=test2
^-- SC2034 (warning): bb appears unused. Verify use (or export if used externally).

For more information:
  https://www.shellcheck.net/wiki/SC2034 -- aa appears unused. Verify use (or...
```

#### Here's what I wanted or expected to see:

```
$ shellcheck -x test.sh 
In include.sh line 3:
bb=test2
^-- SC2034 (warning): bb appears unused. Verify use (or export if used externally).
```

**Comment by koalaman** [comment]
Indeed! This is supported via `-a`:

    -a                  --check-sourced            Include warnings from sourced files
